File: payroll-api/src/main/java/com/company/payroll/mapper/StaffDetailsMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffDetails;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffDetailsMapper {
    int countActiveDepartmentEmployee(Integer deptNo);
    int deleteByPrimaryKey(Integer eId);
    int insert(StaffDetails row);
    int insertSelective(StaffDetails row);
    StaffDetails selectByPrimaryKey(Integer eId);
    List<StaffDetails> selectList();
    int updateByPrimaryKey(StaffDetails row);
    int updateByPrimaryKeySelective(StaffDetails row);
    /**
     * @deprecated This method will be removed after this, since it's no more usable for the database.
     */
    StaffDetails selectByResignId(Integer resignId);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffDetailsMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffDetails
import org.springframework.stereotype.Repository

@Repository
interface StaffDetailsMapper {
  fun deleteByPrimaryKey(staffId: Int): Int

  fun insert(row: StaffDetails): Int

  fun insertSelective(row: StaffDetails): Int

  fun selectByPrimaryKey(staffId: Int): StaffDetails?

  fun updateByPrimaryKeySelective(row: StaffDetails): Int

  fun updateByPrimaryKey(row: StaffDetails): Int
}